installers/msi-language/ActiveState/FileAssociation.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Win32;
+using System;
+
+namespace ActiveState
+{
+
+    public class FileAssociation
+    {
+        // needed so that Explorer windows get refreshed after the registry is updated
+        [System.Runtime.InteropServices.DllImport("Shell32.dll")]
+        private static extern int SHChangeNotify(int eventId, int flags, IntPtr item1, IntPtr item2);
+
+        private const int SHCNE_ASSOCCHANGED = 0x8000000;
+        private const int SHCNF_FLUSH = 0x1000;
+
+        public string Extension { get; set; }
+        public string ProgId { get; set; }
+        public string FileTypeDescription { get; set; }
+        public string ExecutableFilePath { get; set; }
+
+        public bool SetAssociation()
+        {
+            bool madeChanges = false;
+            madeChanges |= SetKeyDefaultValue(@"Software\Classes\" + Extension, ProgId);
+            madeChanges |= SetKeyDefaultValue(@"Software\Classes\" + ProgId, FileTypeDescription);
```

Comment:
When I was implementing this in the `wxs` file I didn't find that this registry key was necessary. What does this one do?

Reply:
It shows up here: ![PerlDocument](https://user-images.githubusercontent.com/8608210/88342787-f0aa4a80-ccf4-11ea-8cc7-0a6f18522fae.png)